src/ring_theory/polynomial/hermite.lean

```diff
@@ -50,4 +51,71 @@ begin
   simp only [map_one, mul_one, derivative_one, sub_zero]
 end
 
+section coeff
+
+lemma coeff_hermite_succ_zero (n : ℕ) :
+  coeff (hermite (n + 1)) 0 = -(coeff (hermite n) 1) := by simp [coeff_derivative]
+
+lemma coeff_hermite_succ_succ (n k : ℕ) :
+  coeff (hermite (n + 1)) (k + 1) = coeff (hermite n) k - (k + 2) * (coeff (hermite n) (k + 2)) :=
+begin
+  rw [hermite_succ, coeff_sub, coeff_X_mul, coeff_derivative, mul_comm],
+  norm_cast
+end
+
+lemma hermite_coeff_upper (n k : ℕ) : coeff (hermite n) (n + k + 1) = 0 :=
+begin
+  induction n with n ih generalizing k,
+  { apply coeff_C },
+  {  rw [coeff_hermite_succ_succ,
+      (by linarith : n + 1 + k = n + k + 1),
+      (by linarith : n + k + 1 + 2 = n + (k + 2) + 1),
+      ih k, ih (k + 2), mul_zero, sub_zero] }
+end
+
+lemma hermite_coeff_upper' (n k : ℕ) (hnk : n < k) : coeff (hermite n) k = 0 :=
+begin
+  obtain ⟨k, rfl⟩ := nat.exists_eq_add_of_lt hnk,
+  apply hermite_coeff_upper
+end
+
+lemma coeff_hermite_self (n : ℕ) : coeff (hermite n) n = 1 :=
+begin
+  induction n with n ih,
+  { apply coeff_C },
+  { rw [coeff_hermite_succ_succ, ih, hermite_coeff_upper, mul_zero, sub_zero] }
+end
+
+@[simp] lemma degree_hermite (n : ℕ) : (hermite n).degree = n :=
+begin
+  rw degree_eq_of_le_of_coeff_ne_zero,
+  simp_rw [degree_le_iff_coeff_zero, with_bot.coe_lt_coe],
+  exact hermite_coeff_upper' n,
+  simp [coeff_hermite_self n]
+end
+
+@[simp] lemma nat_degree_hermite {n : ℕ} : (hermite n).nat_degree = n :=
+nat_degree_eq_of_degree_eq_some (degree_hermite n)
+
+@[simp] lemma leading_coeff_hermite (n : ℕ) : (hermite n).leading_coeff = 1 :=
+begin
+  rw [← coeff_nat_degree, nat_degree_hermite, coeff_hermite_self],
+end
+
+lemma hermite_monic {n : ℕ} : (hermite n).monic := leading_coeff_hermite n
+
+lemma coeff_hermite_of_odd_add {n k : ℕ} (hnk : odd (n + k)) : coeff (hermite n) k = 0 :=
+begin
+  induction n with n ih generalizing k,
```

Comment:
Can't you deduce this from `hermite n (-x) = (-1)^n • hermite n x` and [polynomial.coeff_neg](https://leanprover-community.github.io/mathlib_docs/data/polynomial/basic.html#polynomial.coeff_neg)?

Reply:
I found it easier to prove this (`coeff_hermite_of_odd_add`) and use it later to show `hermite n (-x) = (-1)^n • hermite n x` using the following lemmas:
```
lemma polynomial_even (hp : ∀ n, odd n → p.coeff n = 0) (x) : p.eval (-x) = p.eval x := sorry

lemma polynomial_odd (hp : ∀ n, even n → p.coeff n = 0) (x) : p.eval (-x) = -(p.eval x) := sorry
```

Reply:
````
lemma hermite_comp_neg_X (n : ℕ) : (hermite n).comp (-X) = (-1)^n * hermite n :=
begin
  induction n  with n ih,
  { simp only [hermite_zero, eq_int_cast, algebra_map.coe_one, one_comp, pow_zero, mul_one] },
  have := (hermite n).derivative_comp (-X),
  -- equivalvent to `simp [ih, pow_succ, polynomial.derivative_pow] at this ⊢,`
  simp only [ih, pow_succ, polynomial.derivative_pow, hermite_succ, sub_comp, mul_comp, X_comp,
    neg_mul, one_mul, derivative_mul, derivative_neg, derivative_one, neg_zero, mul_zero,
    zero_mul, zero_add, derivative_X] at this ⊢,
  rw [sub_eq_add_neg, ← this],
  ring,
end
````
To conclude the properties of the coefficients, we need a variant of [polynomial.coeff_comp_degree_mul_degree](https://leanprover-community.github.io/mathlib_docs/data/polynomial/eval.html#polynomial.coeff_comp_degree_mul_degree) for `q = -X`.

Reply:
@lukemantle 's lemma also implies vanishing of coefficients in characteristic two (although `hermite` is currently defined over the integers)

Reply:
Given that we already have the existing proof of `coeff_hermite_of_odd_add`, is it prefereable to use this alternate method involving `hermite_comp_neg_X`? Or is `coeff_hermite_of_odd_add` good enough as-is?

Reply:
I don't have the final say in that, but I think my proposal has the advantage that it is conceptually clearer and needs less code that is specific for Hermite polynomials (I would also say that adding a missing lemma about composing with `-X` is a feature).

Reply:
I think the current lemma is fine. The lemma about composing with X is certainly good to have too, but I assume @lukemantle is planning to add lemmas about `eval` in a subsequent PR.

Reply:
I don't have anything that builds off of `eval` in particular; this was just the way I found easiest to express the even/odd properties of `hermite n`. I did have a go with the method involving composition with `X`, but I was having some difficulty with that, so I was wondering how much of a preference there is between the two methods.

Reply:
Here is one possibility modeled on `polynomial.coeff_comp_degree_mul_degree`.
```
lemma polynomial.coeff_comp_neg_X {R : Type} [comm_ring R] {p : polynomial R} (k : ℕ) :
  (p.comp (-X)).coeff k = (-1)^k * (p.coeff k) :=
begin
  rw [mul_comm, comp, eval₂, coeff_sum, (by simp : -(X : polynomial R) = C (-1) * X)],
  simp_rw [coeff_C_mul, mul_pow, ← C_pow, coeff_C_mul_X_pow],
  convert finset.sum_eq_single k _ _,
  { rw if_pos rfl },
  { assume b hbp hbk,
    rw [if_neg hbk.symm, mul_zero] },
  { assume hkp,
    rw [not_mem_support_iff.mp hkp, zero_mul] }
end
```

Reply:
That said, while this is certainly a nice alternative way to prove it, either way something has to be proven about Hermite polynomials in particular. I don't personally feel strongly about one approach being better.

Reply:
I think I'd like to go ahead with the `eval` version for this PR, if that's ok, and the composition with `X` could be added in a later PR.

Reply:
Did you mean to say "the `coeff` version"?

Reply:
Ah yes, I meant the current version:
```
lemma coeff_hermite_of_odd_add {n k : ℕ} (hnk : odd (n + k)) : coeff (hermite n) k = 0 :=
begin
  induction n with n ih generalizing k,
  { rw zero_add at hnk,
    exact coeff_hermite_of_lt hnk.pos },
  { cases k,
    { rw nat.succ_add_eq_succ_add at hnk,
      rw [coeff_hermite_succ_zero, ih hnk, neg_zero] },
    { rw [coeff_hermite_succ_succ, ih, ih, mul_zero, sub_zero],
      { rwa [nat.succ_add_eq_succ_add] at hnk },
      { rw (by rw [nat.succ_add, nat.add_succ] : n.succ + k.succ = n + k + 2) at hnk,
        exact (nat.odd_add.mp hnk).mpr even_two }}}
end
```
but I ended up saying `eval` because I was thinking about how this version corresponds to a possible later PR where I show the `eval` equivalent of `hermite n (-x) = (-1)^n • hermite n x`.